packages/augur-tools/src/libs/contract-api.ts

```diff
@@ -274,41 +276,10 @@ export class ContractAPI {
     await this.augur.contracts.trade.publicFillBestOrder(type, marketAddress, outcome, numShares, price, tradeGroupID, new BigNumber(3), formatBytes32String(''));
   }
 
-  async takeOrders(orders: ZeroXOrder[]) {
-    const zxOrders = [];
-    const signatures = [];
+  async takeOrders(orders: NativePlaceTradeDisplayParams[]) {
     orders.forEach((order) => {
-      zxOrders.push({
-        makerAddress: order.makerAddress,
-        takerAddress: order.takerAddress,
-        feeRecipientAddress: order.feeRecipientAddress,
-        senderAddress: order.senderAddress,
-        makerAssetAmount: order.makerAssetAmount,
-        takerAssetAmount: order.takerAssetAmount,
-        makerFee: order.makerFee,
-        takerFee: order.takerFee,
-        expirationTimeSeconds: order.expirationTimeSeconds,
-        salt: order.salt,
-        makerAssetData: order.makerAssetData,
-        takerAssetData: order.takerAssetData,
-        makerFeeAssetData: order.makerFeeAssetData,
-        takerFeeAssetData: order.takerFeeAssetData,
-      });
-      signatures.push(order.signature);
+      this.augur.placeTrade(order);
```

Comment:
The promises here aren't being handled. This may be why there are unhandled exception errors.

Reply:
I was thinking fire and forget. Does it need to be await’ed?

Reply:
I'd bundle the promises together and return Promise.all(...) on them. The caller of this function can decide if they want to await or fire-and-forget. For automated tests it's especially good to await because you want to make sure the state is where it should be when you verify it.

Reply:
Ok, I’ll make that change, just in case tests pick this up. Now that ai think contract-api prob already has this functionality. Since I changed it to use augur placeTrade